filters/apiusagemonitoring/spec.go

```diff
@@ -290,58 +286,61 @@ func (s *apiUsageMonitoringSpec) buildClientTrackingInfo(apiIndex int, api *apiC
 	}
 }
 
-// generateRegExpStringForPathTemplate normalizes the given path template and
-// creates a regular expression from it.
-func generateRegExpStringForPathTemplate(pathTemplate string) (normalizedPathTemplate, matcher, pathLabel string) {
-	pathParts := strings.Split(pathTemplate, "/")
-	matcherPathParts := make([]string, 0, len(pathParts))
-	normalizedPathTemplateParts := make([]string, 0, len(pathParts))
-	pathLabelParts := make([]string, 0, len(pathParts))
-	for _, p := range pathParts {
-		if p == "" {
-			continue
-		}
-		placeholderName := getVarPathPartName(p)
-		if placeholderName == "" {
-			// this part is not a placeholder: match it exactly
-			matcherPathParts = append(matcherPathParts, p)
-			normalizedPathTemplateParts = append(normalizedPathTemplateParts, p)
-			pathLabelParts = append(pathLabelParts, p)
-
-		} else {
-			// this part is a placeholder: match a wildcard for it
-			matcherPathParts = append(matcherPathParts, regexUrlPathPart)
-			normalizedPathTemplateParts = append(normalizedPathTemplateParts, ":"+placeholderName)
-			pathLabelParts = append(pathLabelParts, "{"+placeholderName+"}")
-		}
-	}
-	rawRegEx := &strings.Builder{}
-	rawRegEx.WriteString("^")
-	rawRegEx.WriteString(regexOptionalSlashes)
-	rawRegEx.WriteString(strings.Join(matcherPathParts, `\/`))
-	rawRegEx.WriteString(regexOptionalSlashes)
-	rawRegEx.WriteString("$")
+var (
+	regexpMultipleSlashes   = regexp.MustCompile(`/+`)
+	regexpLeadingSlashes    = regexp.MustCompile(`^/*`)
+	regexpTrailingSlashes   = regexp.MustCompile(`/*$`)
+	regexpMiddleSlashes     = regexp.MustCompile(`([^/^])/+([^/*])`)
+	rexexpSlashColumnVar    = regexp.MustCompile(`/:([^:{}/]*)`)
+	rexexpCurlyBracketVar   = regexp.MustCompile(`{([^{}]*?)([?]?)}`)
```

Comment:
the question mark after the star is not needed, if I am not mistaken

Reply:
In this case it is, because else the match would change in an undesired way. E.g. `/resource/{a}-{b}`using `{([^{}]*?)([?]?)}` creates two matches `{a}` and `{c}`, while `{([^{}]*)([?]?)}` would result in a greedy behavior matching only once `{a}-{b}`.